codebase/superdarn/src.bin/tk/tool/fit_speck_removal.1.0/fit_speck_removal.c

```diff
@@ -0,0 +1,364 @@
+  }
+  do {
+  
+    if (vb) {
+      fprintf(stderr,"%.4d-%.2d-%.2d %.2d:%.2d:%.2d %.2d %.2d\n",
+              prm->time.yr,prm->time.mo,prm->time.dy,
+              prm->time.hr,prm->time.mt,prm->time.sc,
+              prm->channel,prm->bmnum);
+    }
+    
+    beam=prm->bmnum;
+    channel=prm->channel;
+    for (int range = 0; range<prm->nrang; range++) {
+      
+      if (fit->rng[range].qflg==1) {
+        index_list[0]=get_index(beam,channel,range  ,irec[beam][channel]  ,maxbeam,maxchannel,maxrange);
+        index_list[1]=get_index(beam,channel,range-1,irec[beam][channel]  ,maxbeam,maxchannel,maxrange);
+        index_list[2]=get_index(beam,channel,range+1,irec[beam][channel]  ,maxbeam,maxchannel,maxrange);
+        index_list[3]=get_index(beam,channel,range,  irec[beam][channel]-1,maxbeam,maxchannel,maxrange);
+        index_list[4]=get_index(beam,channel,range-1,irec[beam][channel]-1,maxbeam,maxchannel,maxrange);
+        index_list[5]=get_index(beam,channel,range+1,irec[beam][channel]-1,maxbeam,maxchannel,maxrange);
+        index_list[6]=get_index(beam,channel,range,  irec[beam][channel]+1,maxbeam,maxchannel,maxrange);
+        index_list[7]=get_index(beam,channel,range-1,irec[beam][channel]+1,maxbeam,maxchannel,maxrange);
+        index_list[8]=get_index(beam,channel,range+1,irec[beam][channel]+1,maxbeam,maxchannel,maxrange);
+        
+        // corners
+        if (irec[beam][channel]==0 && range==0)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[2]] 
+              + 2*qflgs->value[index_list[6]] + qflgs->value[index_list[8]];
+        else if (irec[beam][channel]==0 && range==maxrange-1)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[6]] + qflgs->value[index_list[7]];
+        else if (irec[beam][channel]==nrec[beam][channel] && range==0)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[2]] 
+              + 2*qflgs->value[index_list[3]] + qflgs->value[index_list[5]];
+        else if (irec[beam][channel]==nrec[beam][channel] && range==maxrange-1)
+          sum = 3*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[3]] + qflgs->value[index_list[4]];
+        
+        // edges
+        else if (irec[beam][channel]==0)
+          sum = 2*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[2]] + qflgs->value[index_list[6]] 
+              + qflgs->value[index_list[7]]   + qflgs->value[index_list[8]];
+        else if (irec[beam][channel]==nrec[beam][channel])
+          {sum = 2*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
+              + 2*qflgs->value[index_list[2]] + qflgs->value[index_list[3]] 
+              + qflgs->value[index_list[4]]   + qflgs->value[index_list[5]];
+              fprintf(stderr,"%d %d\n",irec[beam][channel],nrec[beam][channel]);
+              }
```

Comment:
Just deleting some test code I forgot about
```suggestion
          sum = 2*qflgs->value[index_list[0]] + 2*qflgs->value[index_list[1]] 
              + 2*qflgs->value[index_list[2]] + qflgs->value[index_list[3]] 
              + qflgs->value[index_list[4]]   + qflgs->value[index_list[5]];
```